web/html/forms-ii/radio-buttons-ii.md

```diff
@@ -0,0 +1,113 @@
+---
+author: Stefan-Stojanovic
+
+levels:
+  - beginner
+  - basic
+
+type: normal
+
+category: must-know
+
+aspects:
+  - introduction
+  - workout
+
+links:
+  - '[HTML Radio Button](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio){documentation}'
+
+---
+# Creating Radio Buttons
+---
+## Content
+
+When creating radio buttons, each button within the same radio group is given the same `name`.
+
+Example:
+```html
+<form>
+ <p>
+   Please select your gender:
+ </p>
+  <div>
+   <input
+      type="radio" id="gender1"
+      name="gender" value="male">
+   <label
+      for="gender1">Male
+   </label>
+
+   <input
+      type="radio" id="gender2"
+      name="gender" value="female">
+   <label
+      for="gender2">Female
+   </label>
+
+   <input
+      type="radio" id="gender3"
+      name="gender" value="other">
+   <label
+      for="gender3">Other
+   </label>
+  </div>
+  <div>
+    <button
+      type="submit">Submit
+    </button>
+  </div>
+</form>
+```
+
+![radio-empty](https://img.enkipro.com/bb20ad9b34393a58aed6266ebef4144e.png)
+
+[View CodePen](https://codepen.io/enkidevs/pen/QBdKZJ)
+
+In the example above, we have 3 `radio` buttons, each with the same name - `gender` and each with a unique `value` that is used to identify each button. Also, each button has an `id`. This `id` is used with the `<label>` element's `for` attribute to link the label to the radio button.
+
+**Note: By adding the label, we allow the user to have a larger clickable area for the radio button. That is, if the user presses anywhere in the `gender1` label, that label's button will be selected. This option is really useful as it makes it easier to click both for big and small screen sizes.**
+
+Once a user presses a button, and submits the result, the form's data is sent in a `name/value` pair. For instance, if they pick `female`, the form's data will include the line `gender=female` upon submission.
+
+On the other hand, if no value is specified, and the user selects the same button, the result will be sent in the form of `gender=on`, and since this isn't really helpful, you should always include the `value` attribute. Similarly, if no button is selected, the form won't submit at all, since it has no value to report.
+
+---
+## Practice
+
+Write the form code that creates a radio button.
+
+```html
+<??? ???="???"
+     name="topic"
+     value="item"> Item
+```
+
+* input
+* type
+* radio
+* name
+* value

```

Comment:
@Stefan-Stojanovic what's with the last `Item` as in:

```
value="item"> Item
```

is it supposed to be there?

Reply:
Yes.
With `item`:
[with](https://img.enkipro.com/be512adb6ee2b33701dfb6b058b4fe65.png)
Without `item`:
[without](https://img.enkipro.com/db787ed71bbb8e6ceb3f9035295343cc.png)

Reply:
so shouldn't we be using a `<label>` element?

Reply:
I can't find anything about this syntax. I can see it works, but everyone is using `<label>`s and I think we should teach using that as well

Reply:
@kapnobatai137 this isn't really special syntax, it's exactly what it looks like, putting text next to an input element.

Wrapping  a `label` around the text gives more structure and allows some built-in form controls as well as usage CSS selectors.

[Advantages of `label`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label).

I would suggest that we use the `label` to, but this syntax here technically works too :)

Reply:
I 100% agree that it works. I just found it confusing, mostly because I knew that `<input />` is a void element, so it couldn't have any content. 

This syntax also works:
```
    <input type="radio" name="topic" value="item">Item
    <input type="radio" name="topic" value="foobar">Foobar
```

Which increased the confusion :sweat_smile:

My suggestion for this is to simply remove `Item`, as the questions will be exactly the same without it (this would apply to both the Practice Question and the Revision Question)

Reply:
Makes sense 👍